src/main/java/edu/ucsd/sbrg/escher/converter/Escher2Standard.java

```diff
@@ -307,7 +303,7 @@ private void preProcessReaction(EscherReaction reaction,
       Segment currSegment = reaction.getSegment(curve.get(0));
       boolean isProduct = metabolite.getCoefficient() > 0d;
       Node currNode = escherMap.getNode(isProduct ? currSegment.getFromNodeId() : currSegment.getToNodeId());
-      while (!currNode.isMidmarker()) {
+      while (!currNode.isMidmarker() && (currNode.getConnectedSegments().size() != 1 || currNode.isMetabolite())) {
```

Comment:
As a general note: It is often recommended to use more parentheses for unambigious indication of logical expressions.